### DISCUSSION TITLE
Level-set, X-FEM or subdomains for different physic model

### DISCUSSION THREAD

**Discussion by Shaoqigit** [mfem/mfem#2155] [Ideas]
Dear mfem developers,

I would like to know if you have any plan to implement X-FEM with level-set function in the project in near future, or some class to facilitate assignment of the different physic models in different subdomains.
I see that you have similar discussions for this request and proposed some solutions. I think that X-FEM with level-set function may be another solution to ease the subdivision of the domain with different models, it is also a good tool to model some discontinuity problems.

Look forward to your response and thanks for this wonderful FEM project

**Comment by v-dobrev** [comment]
Hi @Shaoqigit, at this point I'm not aware of specific plans to implement X-FEM in MFEM.

Something related that is being currently developed is this PR: https://github.com/mfem/mfem/pull/2043 and the existing miniapp(s) in https://github.com/mfem/mfem/tree/master/miniapps/shifted -- @vladotomov, @kmittal2, or @bslazarov can provide more details, if you are interested.
